### PR TITLE
Added Dashboard tiles

### LINKED ISSUE #147: [FRONTED] Integrate the Dashboard tiles
Fetch data from API to show on the Dashboard tiles.

### PR DESCRIPTION
**Pull request by Anirban-1490**
fixes #147 

Added dashboard data for Total views , Views , Total links created and links active

Screenshot - 

sssss

### CODE REVIEW

**Review by BRAVO68WEB — CHANGES_REQUESTED** [review]
Revert

**Comment by Anirban-1490** [comment]
> Revert

meaning ?

**Comment by BRAVO68WEB** [comment]
Why did you add the vercel testing/preview link to cors

**Comment by Anirban-1490** [comment]
> Why did you add the vercel testing/preview link to cors

oh ok i shoud be removing it.

**Comment by Anirban-1490** [comment]
> Why did you add the vercel testing/preview link to cors

ok removed it can you check now?

**Comment by BRAVO68WEB** [comment]
@Palak-137

**Comment by Palak-137** [comment]
The functionalities of the project have not been working properly because of a CORS error, please check and revert back.

**Comment by Anirban-1490** [comment]
> The functionalities of the project have not been working properly because of a CORS error, please check and revert back. 

Ok looking into it

**Comment by Anirban-1490** [comment]
> The functionalities of the project have not been working properly because of a CORS error, please check and revert back.

ok now see it. it's working on my localhost btw

**Comment by BRAVO68WEB** [comment]
@Palak-137 Are you still reviewing?

**Comment by Palak-137** [comment]
> @Palak-137 Are you still reviewing?

it still has some cors error in production mode

**Comment by Anirban-1490** [comment]
> > @Palak-137 Are you still reviewing?
> 
> it still has some cors error in production mode

but how can i check that? btw i have used the same code that was there for CORS when i have forked this repo  first

**Comment by BRAVO68WEB** [comment]
Let me take a look

File: frontend/components/Dashboard/Dashboard.jsx [hunk 1]
```diff
@@ -57,6 +62,24 @@ function Sidebar1() {
...
 
 function Sidebar2() {
+
+ 
+
+
+
+
+    function fetchTotalLinks(){
+        return axios.get(`http://localhost:5000/minify/all`)
```
Comment (BRAVO68WEB) [inline frontend/components/Dashboard/Dashboard.jsx:72]: Use the env variable to fetch the API url

File: frontend/components/Dashboard/Dashboard.jsx [hunk 2]
```diff
@@ -90,6 +131,23 @@ function Graph() {
     categories: [1991, 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999],
     data: [30, 40, 45, 50, 49, 60, 70, 91],
   })
+
+
+  const context = useContext(UserAuth)
+  function fetchTotalUserLinks(){
+      return axios.get(`http://localhost:5000/minify/all/user`,{
```
Comment (BRAVO68WEB) [inline frontend/components/Dashboard/Dashboard.jsx:138]: Use the env var to fetch API url

File: frontend/components/HomeSection/HomeSection.jsx [hunk 3]
```diff
@@ -81,9 +84,20 @@ function HomeSection(props) {
...
+       if(!context.jwt){
         res = await Axios.post(`/minify/add`, {
-          originalUrl: props.longUrl,
-        })
+            originalUrl: props.longUrl,
+          })
+       }
+       else{
+
+        res = await Axios.post(`/minify/add/custom`, {
```
Comment (BRAVO68WEB) [inline frontend/components/HomeSection/HomeSection.jsx:94]: Or use may use this Axios helper already made instead of env variable

**Review by BRAVO68WEB — CHANGES_REQUESTED** [review]
@Anirban-1490 Do these changes

File: frontend/components/Dashboard/Dashboard.jsx (hunk 1 above)
Comment (Anirban-1490) [reply frontend/components/Dashboard/Dashboard.jsx:72]: > Use the env variable to fetch the API url

or i can use "/minify/all" also? another thing how can i access this #NEXT_PUBLIC_API_URL from the .env . it's giving me undefined

Comment (Anirban-1490) [reply frontend/components/Dashboard/Dashboard.jsx:72]: > Use the env variable to fetch the API url

i have changed this line with axios helper method
Capture1

File: frontend/components/Dashboard/Dashboard.jsx (hunk 2 above)
Comment (Anirban-1490) [reply frontend/components/Dashboard/Dashboard.jsx:138]: > Use the env var to fetch API url

same with this one
Capture

File: frontend/components/Dashboard/Dashboard.jsx (hunk 1 above)
Comment (BRAVO68WEB) [reply frontend/components/Dashboard/Dashboard.jsx:72]: yeah use this

File: frontend/components/Dashboard/Dashboard.jsx (hunk 2 above)
Comment (BRAVO68WEB) [reply frontend/components/Dashboard/Dashboard.jsx:138]: https://github.com/BRAVO68WEB/url-minify/pull/150#discussion_r867492134

File: frontend/components/Dashboard/Dashboard.jsx (hunk 1 above)
Comment (Anirban-1490) [reply frontend/components/Dashboard/Dashboard.jsx:72]: > yeah use this

ok doing the commit

**Comment by Anirban-1490** [comment]
> @Anirban-1490 Do these changes

done. check it

**Review by BRAVO68WEB — APPROVED** [review]
Ok

**Comment by Anirban-1490** [comment]
> Ok

OMG FINALLY

**Pull request merged**